### PR TITLE
[Monitoring] Add metric alerts api 2018-03-01

### PR DESCRIPTION
**Pull request by LeoVannelli**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by olydis** [comment]
@LeoVannelli this went through the API review board now, correct? 🙂

**Comment by olydis** [comment]
@ravbhatnagar new API

**Comment by LeoVannelli** [comment]
No, we started with a preview api version and it got stuck in review, but since we are not going to release the preview version I abandoned the previous pull request and updated it to the GA release version with the minor changes required.

It is based on the preview version in most ways except this one defines the explicit types rather than being weakly typed.

-Leo

**Comment by rboucher** [comment]
@rboucher. Please notify me when this PR is accepted.  Docs are waiting on it.

**Comment by snehithm** [comment]
@ravbhatnagar Can you review this? This is the GA version of metricalerts API. Preview version was already reviewed by you. @LeoVannelli added information above about the deltas.

**Comment by lmazuel** [comment]
@AutorestCI rebuild azure-sdk-for-python

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: can we make odata.type as an enum and then set modelAsString to true?

**Review by ravbhatnagar — COMMENTED** [review]
Just one comment. Other than that looks good.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side. Feel free to merge based on the above comment.

**Comment by lmazuel** [comment]
@AutorestCI regenerate azure-sdk-for-python

**Comment by olydis** [comment]
@LeoVannelli thanks for your response regarding API review board. You mentioned the preview version got "stuck" in review - but what about this version? Has it been reviewed by the board or what is the plan? 🙂

**Comment by LeoVannelli** [comment]
The review starts in 12 min 😊 but I don’t see many people accepted the review.

-Leo

**Comment by LeoVannelli** [comment]
Pull request has been updated with review feedback. Review board has approved. Waiting on final review and merge.

**Comment by snehithm** [comment]
@olydis Leo had a review and it was signed off in the meeting. Can we get this merged? We are waiting on this merge to go through for us update the docs.

Regards,
Snehith

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: `"operationId": "MetricAlertsStatus_listByName",`

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: `"description": "criteria"` is really vague and it causes a [validation error](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/356119128#L731) -- can you put in a better description?

**Comment by fearthecowboy** [comment]
You have a couple of small typos (mentioned in comments)

and your example data isn't passing validation (see: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/356119129#L733  )

**Comment by LeoVannelli** [comment]
I am trying to understand the validation report but it is not clear what it is doing? How do I know what failed? And what it was validating against when it failed?

-Leo

**Comment by LeoVannelli** [comment]
Updated examples with corrections, not sure if all items are fixed, how can I run the test you executed to validate locally

**Comment by fearthecowboy** [comment]
Hey @LeoVannelli  

You can run the tests locally by [installing autorest ](https://aka.ms/autorest/install) and then running:

``` powershell
# start where your readme.md file is.
cd azure-rest-api-specs/specification/monitor/resource-manager

# run the linter rules validation
autorest --azure-validator readme.md

# install the oav tool  (example validation)
npm install -g oav

# run the oav tool 
oav validate-example microsoft.insights/stable/2018-03-01/metricAlert_API.json

```

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json
Comment (RikkiGibson) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json]: *defines

**Comment by LeoVannelli** [comment]
Corrected.

**Comment by LeoVannelli** [comment]
Any updates on this? Examples have been fixed just waiting on approval and merge.

**Review by fearthecowboy — APPROVED** [review]

**Pull request merged**
